.github/workflows/ci.yml

```diff
@@ -0,0 +1,42 @@
+name: SDK Rust CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  test:
+    name: cargo test
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+        rust: [stable, nightly]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: ${{ matrix.rust }}
+      - name: Test
+        run: cargo test --workspace
```

Comment:
Do you think it's worth adding a `cargo build` to make sure our builds are passing as well? (Or is it embedded in the `cargo test` command already?)

Reply:
I went ahead and did a little digging here to find out the specifics:
* `cargo test` will implicitly run `cargo build`, BUT it will also build code that is specifically marked with `#[cfg(test)]` or `#[test]` (in addition to actually running the tests themselves).

Our upcoming bindings code will rely on artifacts generated via `cargo build`, as we don't want to include test code as a part of our bindings. I'll go ahead and add this in!